packages/inventory/src/ducks/inventory.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  fetchActionCreatorFactory,
+  removeActionCreatorFactory,
+  setTotalRecordsFactory,
+  reducerFactory,
+  getItemsArrayFactory,
+  getTotalRecordsFactory,
+} from '@opensrp/reducer-factory';
+import { values } from 'lodash';
+import { Store } from 'redux';
+import { createSelector } from 'reselect';
+import { Dictionary } from '@onaio/utils';
+
+/** interface inventory */
+export interface Inventory {
+  deliveryDate: string;
+  donor: string;
+  poNumber: number;
+  productName: string;
+  providerId: string;
+  quantity: number;
+  serialNumber: string;
+  servicePointId: string;
+  stockId: string;
+  unicefSection: string;
+}
+
+/** reducer name */
+export const inventoryReducerName = 'inventory';
+
+/** Item Reducer */
+export const inventoryReducer = reducerFactory<Inventory>(inventoryReducerName);
+
+// action
+/** actionCreator returns action to to add Item records to store */
+export const fetchInventories = fetchActionCreatorFactory<Inventory>(
+  inventoryReducerName,
+  'servicePointId'
```

Comment:
I thought about this, and I think we should use the stockIds here instead of service pointId